### ISSUE TITLE
5 patches resized to  the same size?  

### ISSUE THREAD

**Issue by tengshaofeng** [chenzeyuczy/DeepID2#1]
Hi chenzeyuczy,
    I am so appreciated with your great job. I have read your code today.
but i have a question about following code:

# code start
#PATCH_INFO=(_0_0_90_90_RGB _0_60_90_150_RGB _60_0_150_90_RGB
#_60_60_150_150_RGB _30_30_120_120_RGB)
PATCH_INFO=(_${HEIGHT}_${WIDTH} )

for SUB_PATCH in ${PATCH_INFO[@]}
do
	#ROOT_FOLDER="/home/chenzeyu/dataset/${DATASET}/${DATASET}${SUB_PATCH}/"
	#FILELIST=./data/sample_mix_${DATASET}.txt
	DB_NAME="/home/chenzeyu/dataset/${DATASET}/${DATASET}${SUB_PATCH}-lmdb/"

	ROOT_FOLDER="/home/chenzeyu/dataset/${DATASET}/${DATASET}-cropped/"
	FILELIST=./data/sample_mix_${DATASET}.txt
# code end

    according to the code SUB_PATCH is _55_47_ , does it mean that all the 5 patches in one face image are resized to 55*47?  .  I have witness the generation process of  _0_0_90_90_RGB folder , but i have never witness  how the _55_47_ folder generate.

   look forward your reply.
   regards
   thanks in advance.

**Comment by chenzeyuczy** [comment]
Hi tengshaofeng,
glad to hear from you.
Various sizes have been tried on this experiment. Also, I've try to crop patches, which are stored in folders with suffix like '_0_0_90_90_RGB', from the original images for data augmentation. The patches are also resized before loading into the network.
In the scripts, only a lmdb database will be generated at the end, which is a folder containing two mdb files. And I've specified the folder name in the script.
`DB_NAME="/home/chenzeyu/dataset/${DATASET}/${DATASET}${SUB_PATCH}-lmdb/"`
I invoke _convert_imageset_set.bin_ in $CAFFE_ROOT/build/tools/ in this script, U may refer to it for further instructions.
Actually, codes above are still in a debug mode, which are still far from neat.
Sorry for reply so late and hope it'll help.
Good luck!

**Comment by tengshaofeng** [comment]
@chenzeyuczy
 thanks for your reply so much. It is debug version, and you have not used optimal 25 patches selected as input of the net like the paper said? u just used 5 patches instead ,right? 

DB_NAME="/home/chenzeyu/dataset/${DATASET}/${DATASET}${SUB_PATCH}-lmdb/" 
do you mean that you generate only a CAISIA_55_47-lmdb from all kinds of pathes for example  '_0_0_90_90_RGB'

**Comment by chenzeyuczy** [comment]
@tengshaofeng 
Yeah, I didn't crop the optimal 25 patches as described in the paper, only 4 patches at the corners and the center one are selected from the original image. In the experiment, each type of patches is generated into a particular lmdb database, which is load to to separate network for training.
During the test phase, patches are cropped from image first and then predict in the the corresponding network. Then we can combine the results for the final answer.

**Comment by tengshaofeng** [comment]
@chenzeyuczy
thanks for reply so much.   combine the results for the final answer? it is through a vote ? or concatenate features from different nets , then train a classifier?

**Comment by chenzeyuczy** [comment]
@tengshaofeng 
Take advantage of whatever strategy u favors. For example, voting seems to be a naive but effective method.
